### DISCUSSION TITLE
Possible Memory Leak in Scrypted

### DISCUSSION THREAD

**Discussion by Jety567** [koush/scrypted#1093] [Q&A]
Hey @koush 

I have updated Scrypted to version 0.55.00 a few days ago and since then my memory is raising until its maximum 6 Gb and then Scrypted restarts 

![PNG-Bild](https://github.com/koush/scrypted/assets/64406763/0019fa57-9191-45a2-a722-6224554d9a42)

I run the Scrypted installation on a Proxmox LXC via Docker 

Until the update it runs like a charm

Here a picture of htop 

<img width="1580" alt="Bildschirmfoto 2023-09-24 um 16 08 11" src="https://github.com/koush/scrypted/assets/64406763/60065ab6-f3f7-4499-9c2a-79bb20b2dce2">

Is there anything I can do?

I have connected 4 Reolink Cameras with 4 MP and 5 TP Link cameras with 2K resolution

Connected via the Reolink or ONVIF Plugin

**Comment by leankuroneko** [comment]
I just checked because seeing my server almost maxed out RAM was weird and I just discovered 18G of RAM is going to the Scrypted container 
![image](https://github.com/koush/scrypted/assets/2634199/9a204821-cb98-421d-ad68-edbcfe251127)

On Scrypted dashboard supposedly the highest memory usage should be 410M for NVR 
![image](https://github.com/koush/scrypted/assets/2634199/37a95763-554d-43d7-ada8-ac58ed955066)

I'm on v 0.68.0 , what could it be causing this?

**Comment by koush** [comment]
please use htop to capture memory usage within the container.

**Reply by leankuroneko** [reply]
Sure! 
![image](https://github.com/koush/scrypted/assets/2634199/dbcb1f2d-5422-4949-84d4-abb3bd6bf8a6)

**Comment by koush** [comment]
I'm not sure what you're doing here, but there's a ton of transcoding processes happening here, and Scrtyped tries very hard to prevent users from doing that.

**Comment by koush** [comment]
did you enable the transcoding extension on your cameras by chance?

**Reply by leankuroneko** [reply]
I'm **very** not sure what I'm doing too I think... 

Oh yes! I did that a couple days ago when I was [recording this](https://github.com/koush/scrypted/issues/522#issuecomment-1848269409) about settings not being kept 

Should I disable that?

**Reply by koush** [reply]
yes, delete that extension and reload rebroadcast plugin.

**Reply by leankuroneko** [reply]
Done! I didn't noticed much change until restarting the NVR plugin, there 10G were back

Then I started viewing the cameras, rewinding and going back to live and noticed RAM usage started going up again so I think I may have derped something in NVR settings 😑

![image](https://github.com/koush/scrypted/assets/2634199/d1057e3f-08aa-4158-8cff-558ecb9e6def)
![image](https://github.com/koush/scrypted/assets/2634199/934fdace-b425-4129-85af-480883331369)

**Reply by koush** [reply]
I think you enabled max compat mode in webrtc plugin maybe.

**Reply by koush** [reply]
need to turn off max compat mode in webrtc.

**Reply by leankuroneko** [reply]
Again you're right, some time ago I remember having trouble with the streaming getting stuck on "loading" and this compat mode seemed like it fixed that, so I left it enabled.

Thank you so much for the patience!
